javascript_test.go

```diff
@@ -0,0 +1,229 @@
+						I18N: map[string]string{
+							"en-US": "The Flow account the tokens will go to",
+						},
+					},
+				},
+				Balance: "",
+			},
+		},
+	},
+}
+
+
+var parsedTemplateScript = &FlowInteractionTemplate{
+	FType:    "InteractionTemplate",
+	FVersion: "1.0.0",
+	ID:       "290b6b6222b2a77b16db896a80ddf29ebd1fa3038c9e6625a933fa213fce51fa",
+	Data: Data{
+		Type:      "script",
+		Interface: "",
+		Messages: Messages{
+			Title: &Title{
+				I18N: map[string]string{
+					"en-US": "Multiply Two Integers",
+				},
+			},
+			Description: &Description{
+				I18N: map[string]string{
+					"en-US": "Multiply two numbers to another",
+				},
+			},
+		},
+		Cadence: "pub fun main(x: Int, y: Int): Int { return x * y }",
+		Arguments: Arguments{
+			"x": Argument{
+				Index: 0,
+				Type:  "Int",
+				Messages: Messages{
+					Title: &Title{
+						I18N: map[string]string{
+							"en-US": "number to be multiplied",
+						},
+					},
+				},
+				Balance: "",
+			},
+			"y": Argument{
+				Index: 1,
+				Type:  "Int",
+				Messages: Messages{
+					Title: &Title{
+						I18N: map[string]string{
+							"en-US": "second number to be multiplied",
+						},
+					},
+				},
+				Balance: "",
+			},
+		},
+	},
+}
+
+
+var ArrayTypeScript = &FlowInteractionTemplate{
+	FType:    "InteractionTemplate",
+	FVersion: "1.0.0",
+	ID:       "290b6b6222b2a77b16db896a80ddf29ebd1fa3038c9e6625a933fa213fce51fa",
+	Data: Data{
+		Type:      "script",
+		Interface: "",
+		Messages: Messages{
+			Title: &Title{
+				I18N: map[string]string{
+					"en-US": "Multiply Numbers",
+				},
+			},
+			Description: &Description{
+				I18N: map[string]string{
+					"en-US": "Multiply numbers in an array",
+				},
+			},
+		},
+		Cadence: "pub fun main(numbers: [Int]): Int { var total = 1; for x in numbers { total = total * x }; return total }",
+		Arguments: Arguments{
+			"numbers": Argument{
+				Index: 0,
+				Type:  "[Int]",
+				Messages: Messages{
+					Title: &Title{
+						I18N: map[string]string{
+							"en-US": "Array of numbers to be multiplied",
+						},
+					},
+				},
+				Balance: "",
+			},
+		},
+	},
+}
+
+var minimumTemplate = &FlowInteractionTemplate{
+	FType:    "InteractionTemplate",
+	FVersion: "1.0.0",
+	ID:       "290b6b6222b2a77b16db896a80ddf29ebd1fa3038c9e6625a933fa213fce51fa",
+	Data: Data{
+		Type:      "script",
+		Interface: "",
+		Cadence: "pub fun main(numbers: [Int]): Int { var total = 1; for x in numbers { total = total * x }; return total }",
+		Arguments: Arguments{
+			"numbers": Argument{
+				Index: 0,
+				Type:  "[Int]",
+			},
+		},
+	},
+}
+func TestJSGenTransaction(t *testing.T) {
+	assert := assert.New(t)
+
+	contents, err := GenerateJavaScript(parsedTemplateTX, "./transfer_token.json", true)
+	assert.NoError(err, "ParseTemplate should not return an error")
+	assert.NotNil(contents, "Parsed template should not be nil")
+	assert.True(strings.Contains(contents, "transferTokens"), "Expected '%s'", "transferTokens")
```

Comment:
In order to have good tests here i would considering using https://github.com/hexops/autogold or a similar library. 

golden files are really good here. You get really good asserts and you dont need to update them yourself if they change. 

NB! However it is very important to do PR review of the changed golden files so that a failure is not blindly accepted.

Reply:
We use autogold in find for lots of tests and it works really well.